mllib-local/pom.xml

```diff
@@ -75,6 +75,11 @@
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>dev.ludovic.netlib</groupId>
+      <artifactId>blas</artifactId>
+    </dependency>
```

Comment:
So if I undestand right, you can remove the `all` artifact below in the profile then? (But keep the profile as empty, with a comment that it's no longer used.)

Reply:
Today, `dev.ludovic.netlib` does `com.github.fommil.netlib.BLAS.getInstance()` to probe whether the native libraries are installed. (This call is in `com.github.fommil:core` and is thus all good regarding licensing.) If we remove the content of the `netlib-lgpl` profile and the dependency on `com.github.fommil:all` then we will break the current behavior of probing for native BLAS implementation through `com.github.fommil:native_ref` and `com.github.fommil:native_system`.

In the future, I do want to add JNI-based prober/wrapper for BLAS, LAPACK, and ARPACK (and which are compliment with licensing as discussed above), but that is not done yet. Then, we will be able to remove the reference to `com.github.fommil:all`.

Reply:
I see, so `all` is still needed, and needs to be behind the profile. While not required for license reasons, do any other of your dependencies only makes sense when paired with the `all` artifact, and so should be in the profile too?

Reply:
No, the only dependency of interest is `com.github.fommil`. The others are junit for testing, and jmh for benchmarking (in the `dev.ludovic.netlib:benchmarks` package only).

Reply:
Dumb question, this all works perfectly fine then without the profile active, without netlib-java `all`? then I think we're OK. If the effect is the same: works without the profile, accelerated with the profile (and presence of OpenBLAS etc) then I think this change is OK.

Reply:
Long story short: yes, it works without the profile active (and is even faster, data in https://github.com/apache/spark/pull/32253#issue-619173915), and is hardware accelerated with the profile or when the Foreign Linker API is active.

Let me document what gets probed and the impact of having the profile in place, I think this is going to clarify many of the questions you've had.

Reply:
As updated in the description, the order in which things are used is the following:

|                                          | BLAS.javaBLAS | BLAS.nativeBLAS |
| ------------------------ | --------------- | ------------------- |
| **with** -Pnetlib-lgpl      | 1. `dev.ludovic.netlib.blas.VectorizedBLAS` (JDK16+, relies on the Vector API, requires `--add-modules=jdk.incubator.vector` on JDK16)<br>2. `dev.ludovic.netlib.blas.Java11BLAS` (JDK11+)<br> 3. `dev.ludovic.netlib.blas.JavaBLAS`<br>4. `dev.ludovic.netlib.blas.NetlibF2jBLAS`, a wrapper for `com.github.fommil:core`   |  1. `dev.ludovic.netlib.blas.NetlibNativeBLAS`, a wrapper for `com.github.fommil:all`<br>2. `dev.ludovic.netlib.blas.ForeignBLAS` (JDK16+, relies on the Foreign Memory API, requires `--add-modules=jdk.incubator.foreign -Dforeign.restricted=warn`) <br>3. fails to load, falls back to BLAS.javaBLAS in `org.apache.spark.ml.linalg.BLAS` |
| **without** -Pnetlib-lgpl | 1. `dev.ludovic.netlib.blas.VectorizedBLAS` (JDK16+, relies on the Vector API, requires `--add-modules=jdk.incubator.vector` on JDK16)<br>2. `dev.ludovic.netlib.blas.Java11BLAS` (JDK11+)<br>3. `dev.ludovic.netlib.blas.JavaBLAS`<br>4. `dev.ludovic.netlib.blas.NetlibF2jBLAS`, a wrapper for `com.github.fommil:core`  | 1. `dev.ludovic.netlib.blas.ForeignBLAS` (JDK16+, relies on the Foreign Memory API, requires `--add-modules=jdk.incubator.foreign -Dforeign.restricted=warn`)<br>2. fails to load, falls back to BLAS.javaBLAS in `org.apache.spark.ml.linalg.BLAS`  |